src/mtcp/mtcp_restart.c

```diff
@@ -745,9 +744,10 @@ unmap_memory_areas_and_restore_vdso(RestoreInfo *rinfo)
     }
 #if defined(__i386__)
     // FIXME: For clarity of code, move this i386-specific code toa function.
-    void *vvar = mtcp_sys_mmap(vvarStart, vvarEnd - vvarStart,
-                               PROT_EXEC | PROT_WRITE | PROT_READ,
-                               MAP_PRIVATE | MAP_ANONYMOUS | MAP_FIXED, -1, 0);
+    void *vvar = mmap_fixed_noreplace(vvarStart, vvarEnd - vvarStart,
+                                      PROT_EXEC | PROT_WRITE | PROT_READ,
+                                      MAP_PRIVATE | MAP_ANONYMOUS | MAP_FIXED,
```

Comment:
The `MAP_FIXED` is redundant, since `map_fixed_noreplace()` will simply remove the `MAP_FIXED`.
We can keep it here, or remove `MAP_FIXED` with the idea that the name `map_fied_noreplace()` already says it all.
I'm open to either option (keeping it as is, or removing it).  Just a thought.

Reply:
I want to rename this function at some point to make it more generic. Let's leave the flag in for now.